src/v/cluster/tests/serialization_rt_test.cc

```diff
@@ -2335,7 +2335,12 @@ SEASTAR_THREAD_TEST_CASE(commands_serialization_test) {
     }
 }
 
+/*
+ * with upgraded abseil some types aren't standard layout and so sync serde is
+ * not available via reflection. in these cases, disable the test.
+ */
 template<typename T>
+requires std::is_standard_layout_v<T>
```

Comment:
can't we just do something like `requires reflection::to_tuple(T{})`? That would be easier to understand IMO.